src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Unix.cs

```diff
@@ -2,108 +2,1037 @@
+                _abortSource.Token.Register(s =>
+                {
+                    var thisRef = (ManagedClientWebSocket)s;
+
+                    WebSocketState state = thisRef._state;
+                    if (state != WebSocketState.Closed && state != WebSocketState.Aborted)
+                    {
+                        thisRef._state = state != WebSocketState.None && state != WebSocketState.Connecting ?
+                            WebSocketState.Aborted :
+                            WebSocketState.Closed;
+                    }
+                }, this);
             }
-        }
 
-        public WebSocketCloseStatus? CloseStatus
-        {
-            get
+            public override void Dispose()
             {
-                throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
+                if (!_disposed)
+                {
+                    _disposed = true;
+                    _keepAliveTimer.Dispose();
+                    _stream?.Dispose();
+                    _client?.Dispose();
+                }
             }
-        }
 
-        public string CloseStatusDescription
-        {
-            get
+            public override WebSocketCloseStatus? CloseStatus => _closeStatus;
+
+            public override string CloseStatusDescription => _closeStatusDescription;
+
+            public override WebSocketState State => _state;
+
+            public override string SubProtocol => _subprotocol;
+
+            public async Task ConnectAsync(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
             {
-                throw new PlatformNotSupportedException(SR.net_WebSockets_UnsupportedPlatform);
+                // Not currently used:
+                // - ClientWebSocketOptions.Credentials
+                // - ClientWebSocketOptions.Proxy
+
+                ClientWebSocket.ThrowIfInvalidState(_state, _disposed, s_validConnectStates);
+                _state = WebSocketState.Connecting;
+
+                // Establish connection to the server
+                Stream stream = null;
+                var client = new TcpClient();
+                try
+                {
+                    // Connect over TCP to the remote server
+                    await client.ConnectAsync(uri.Host, uri.Port).ConfigureAwait(false);
+                    stream = client.GetStream();
+
+                    // Upgrade to SSL if needed
+                    if (uri.Scheme == UriScheme.Wss)
+                    {
+                        var sslStream = new SslStream(stream);
+                        await sslStream.AuthenticateAsClientAsync(
+                            uri.Host, 
+                            options.ClientCertificates, 
+                            SecurityProtocol.AllowedSecurityProtocols, 
+                            checkCertificateRevocation: false).ConfigureAwait(false);
```

Comment:
> Skipping revocation?

I did that because:
- It's the default if you use one of the shorter overloads, and as a result it's what we do in other libs, like SqlClient.
- It's the default with HttpClient, where you only get revocation checking if you set CheckCertificateRevocationList to true.
- Etc.
I have no problem changing it, though, if you say that's the right thing to do.

> Is there no input host verification callback?

You mean a callback for verifying the remote certificate?  A RemoteCertificateValidationCallback delegate can be passed to the SslStream ctor to do manual validation of the server's certificate... but why do you ask?  Isn't the default verification sufficient?